ydb/docs/en/core/reference/ydb-cli/tools-infer.md

```diff
@@ -0,0 +1,159 @@
+# Table schema inference from data files
+
+You can use the `{{ ydb-cli }} tools infer csv` command to generate a `CREATE TABLE` statement from a CSV data file. This can be very helpful when you want to [import](./export-import/import-file.md) data into a database and the table has not been created yet.
+
+Command syntax:
+
+```bash
+{{ ydb-cli }} [global options...] tools infer csv [options...] <input files...>
+```
+
+- `global options` – [global options](commands/global-options.md).
+- `options` – [subcommand options](#options).
+
+To get the most up-to-date information about the command, use the `--help` option:
+
+```bash
+{{ ydb-cli }} tools infer csv --help
+```
+
+## Subcommand options {#options}
+
+Option Name | Description
+---|---
+`-p, --path` | Database path to table that should be created. Default: `table`.
+`--columns` | Explicitly specifies table column names, as a comma-separated list.
+`--gen-columns` | Explicitly indicates that table column names should be generated automatically (column1, column2, ...).
+`--header` | Explicitly indicates that the first row in the CSV contains column names.
+`--rowsalyze` | Number of rows to analyze. 0 means unlimited. Reading will be stopped soon after this number of rows is read. Default: `500000`.
+`--execute` | Execute `CREATE TABLE` request right after generation.
+
+{% note info %}
+
+If none of the options `--columns`, `--gen-columns`, or `--header` are explicitly specified, the following algorithm is used: The first row of the file is taken, and its values are checked for the following conditions:
+
+* The values meet the requirements for column names.
+* The types of these values are different from the data types in the other rows of the file.
+
+If both conditions are met, the values from the first row are used as the table's column names. Otherwise, column names are generated automatically. See the [example](#example-default) below for more details.
+
+{% endnote %}
+
```

Comment:
Add a section explaining the inference algorithm. For instance, what happens if numbers and strings are mixed, how it chooses between float and decimal, how it chooses a proper integer size, what happens if the data has both negative integers and values close to `Max<Uint64>`, can it detect some containers, etc.

A table with all the data types it supports would also be helpful.